sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -278,21 +278,45 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def ansiDateTimeError(e: Exception): SparkDateTimeException = {
+    def extractDateTimeErrorInfo(e: Exception): (String, String, String) = {
+      val errorMessage = e.getMessage
+
+      val pattern = "Invalid value for ([A-Za-z]+) \\(valid values (.+)\\): (.+)".r
+
+      errorMessage match {
+        case pattern(field, range, badValue) =>
+          val unit = field match {
+            case "Year" => "YEAR"
+            case "MonthOfYear" => "MONTH"
+            case "DayOfMonth" => "DAY"
+            case "Hour" => "HOUR"
+            case "Minute" => "MINUTE"
+            case "Second" => "SECOND"
+            case _ => field
```

Comment:
What is this?
Also please also generate the range yourself. E.g. the minus '-' is terribly confusing. '...' works better.

Reply:
Sounds good